File: backend/profile-service/src/main/java/com/example/profileservice/global/entity/BaseTimeEntity.java
package com.example.profileservice.global.entity;

import lombok.Getter;

import jakarta.persistence.Column;
import jakarta.persistence.MappedSuperclass;
import jakarta.persistence.PrePersist;
import jakarta.persistence.PreUpdate;
import java.time.LocalDateTime;

@Getter
@MappedSuperclass
public abstract class BaseTimeEntity {

    @Column(updatable = false)
    private LocalDateTime createdDate;

    private LocalDateTime updatedDate;

    @PrePersist
    public void prePersist() {
        createdDate = LocalDateTime.now();
        updatedDate = null;
    }

    @PreUpdate
    public void preUpdate() {
        updatedDate = LocalDateTime.now();
    }
}

File: backend/point-service/src/main/kotlin/com/example/pointservice/common/domain/BaseTimeEntity.kt
package com.example.pointservice.common.domain

import jakarta.persistence.Column
import jakarta.persistence.MappedSuperclass
import jakarta.persistence.PrePersist
import jakarta.persistence.PreUpdate
import java.time.LocalDateTime

@MappedSuperclass
abstract class BaseTimeEntity {

    @Column(updatable = false)
    var createdDate: LocalDateTime? = null
        protected set

    var updatedDate: LocalDateTime? = null
        protected set

    @PrePersist
    fun prePersist() {
        createdDate = LocalDateTime.now()
        updatedDate = null
    }

    @PreUpdate
    fun preUpdate() {
        updatedDate = LocalDateTime.now()
    }
}
